Managers/FrameMan.cpp

```diff
@@ -1,2651 +1,1391 @@
-//					rotation angle in radians, frame to draw
-// Return value:    None.
-
-	void FrameMan::DrawBitmapPrimitive(int player, Vector start, Entity * pEntity, float rotAngle, int frame)
-	{
-		MOSprite * pMOS = dynamic_cast<MOSprite *>(pEntity);
-		if (pMOS)
-		{
-			BITMAP * pBitmap = pMOS->GetSpriteFrame(frame);
-			if (pBitmap)
-				m_Primitives.push_back(new BitmapPrimitive(player, start, pBitmap, rotAngle));
+			// If we are dealing with split screens, then deal with the fact that we need to draw the player screens to different locations on the final buffer
+			// The position of the current draw screen on the final screen
+			Vector screenOffset;
+
+			if (screenCount > 1) {
+				switch (whichScreen) {
+					case 1:
+						// If both splits, or just VSplit, then in upper right quadrant
+						if ((m_VSplit && !m_HSplit) || (m_VSplit && m_HSplit)) {
+							screenOffset.SetIntXY(GetResX() / 2, 0);
+						} else {
+							// If only HSplit, then lower left quadrant
+							screenOffset.SetIntXY(0, GetResY() / 2);
+						}
+						break;
+					case 2:
+						// Always lower left quadrant
+						screenOffset.SetIntXY(0, GetResY() / 2);
+						break;
+					case 3:
+						// Always lower right quadrant
+						screenOffset.SetIntXY(GetResX() / 2, GetResY() / 2);
+						break;
+					default:
+						// Always upper left corner
+						screenOffset.SetIntXY(0, 0);
+						break;
+				}
+			}
+			// Flash the screen if we're supposed to
+			if (m_FlashScreenColor[whichScreen] != -1) {
+				// If set to flash for a period of time, first be solid and then start flashing slower
+				float left = m_FlashTimer[whichScreen].LeftTillRealTimeLimitMS();
+
+				if (/*left > 500 || */left < 10 || m_FlashTimer[whichScreen].AlternateReal(50)) {
+					// At most, only flash every other frame
+					if (m_FlashedLastFrame[whichScreen]) {
+						m_FlashedLastFrame[whichScreen] = false;
+					} else {
+						// FLASH!
+						rectfill(pDrawScreenGUI, 0, 0, pDrawScreenGUI->w, pDrawScreenGUI->h, m_FlashScreenColor[whichScreen]);
+						// Show that we did indeed flash this frame
+						m_FlashedLastFrame[whichScreen] = true;
+					}
+				}
+				// Make things go into slight slow-mo - DANGER
+				//g_TimerMan.SetOneSimUpdatePerFrame(!m_SloMoTimer.IsPastSimTimeLimit());
+				//g_TimerMan.IsOneSimUpdatePerFrame();
+
+				// Stop with the flash after the designated period
+				if (m_FlashTimer[whichScreen].IsPastRealTimeLimit()) { m_FlashScreenColor[whichScreen] = -1; }
+			}
+
+			// Draw the intermediate draw splitscreen to the appropriate spot on the back buffer
+			if (!m_StoreNetworkBackBuffer) { blit(pDrawScreen, m_pBackBuffer8, 0, 0, screenOffset.GetFloorIntX(), screenOffset.GetFloorIntY(), pDrawScreen->w, pDrawScreen->h); }
+
+			// Add the player screen's effects to the total screen effects list so they can be drawn in post processing
+			if (!IsInMultiplayerMode()) {
+				int occX = g_SceneMan.GetScreenOcclusion(whichScreen).GetFloorIntX();
```

Comment:
Pull this out into a separate private method. It's basically all the screen effect handling, maybe it could even go into one of your sub managers?